File: Clase6/SoftProgSolution/SoftProgNegocio/src/main/java/pe/edu/pucp/softprog/rrhh/bo/AreaBOImpl.java
package pe.edu.pucp.softprog.rrhh.bo;

import pe.edu.pucp.softprog.rrhh.boi.IAreaBO;
import pe.edu.pucp.softprog.rrhh.dao.AreaDAO;
import pe.edu.pucp.softprog.rrhh.impl.AreaImpl;
import pe.edu.pucp.softprog.rrhh.model.Area;

import java.util.List;

public class AreaBOImpl implements IAreaBO{

    private final AreaDAO daoArea;

    public AreaBOImpl(){
        daoArea = new AreaImpl();
    }

    @Override
    public int insertar(Area area) throws Exception {
        validar(area, false);
        return daoArea.insertar(area);
    }

    @Override
    public int modificar(Area area) throws Exception {
        validar(area, true);
        return daoArea.modificar(area);
    }

    @Override
    public int eliminar(int id) throws Exception {
        if (id <= 0) {
            throw new Exception("El id del area debe ser mayor que cero.");
        }
        return daoArea.eliminar(id);
    }

    @Override
    public List<Area> listarTodos() throws Exception {
        return daoArea.listarTodos();
    }

    @Override
    public Area buscarPorId(int id) throws Exception {
        if (id <= 0) {
            throw new Exception("El id del area debe ser mayor que cero.");
        }
        return daoArea.buscarPorId(id);
    }

    private void validar(Area area, boolean esModificacion) throws Exception {
        if (area == null) {
            throw new Exception("El área no puede ser nula.");
        }

        if (esModificacion && area.getIdArea() <= 0) {
            throw new Exception("El id del área es obligatorio para la modificación.");
        }

        validarNombre(area.getNombre());
    }

    private void validarNombre(String nombre) throws Exception {
        if (nombre == null || nombre.trim().isEmpty()) {
            throw new Exception("El nombre del área es obligatorio.");
        }

        nombre = nombre.trim();

        if (nombre.length() < 3) {
            throw new Exception("El nombre del área debe tener al menos 3 caracteres.");
        }

        if (nombre.length() > 30) {
            throw new Exception("El nombre del área no puede exceder los 75 caracteres.");
        }
    }
}

File: Clase12/ProyectosNET/SoftProgSolution/SoftProgNegocio/RRHH/BO/AreaBOImpl.cs
using SoftProgModel.RRHH;
using SoftProgNegocio.RRHH.BOI;
using SoftProgPersistencia.RRHH.DAO;
using SoftProgPersistencia.RRHH.Impl;
using System;
using System.Collections.Generic;
using System.Text;

namespace SoftProgNegocio.RRHH.BO
{
    public class AreaBOImpl : IAreaBO
    {
        private AreaDAO daoArea;
        public AreaBOImpl()
        {
            daoArea = new AreaImpl();
        }
        public Area buscarPorId(int id)
        {
            throw new NotImplementedException();
        }

        public int eliminar(int idObjeto)
        {
            throw new NotImplementedException();
        }

        public int insertar(Area area)
        {
            //validaciones
            return daoArea.insertar(area);
        }

        public List<Area> listarTodos()
        {
            return daoArea.listarTodos();
        }

        public int modificar(Area objeto)
        {
            throw new NotImplementedException();
        }
    }
}
